File: hw2/RaceMonitor_CSharp/RaceData/Messages/RegistrationUpdate.cs
using System;

namespace RaceData.Messages
{
    public class RegistrationUpdate : AthleteUpdate
    {
        public string FirstName { get; set; }
        public string LastName { get; set; }
        public string Gender { get; set; }
        public int Age { get; set; }

        public RegistrationUpdate() : base(AthleteRaceStatus.Registered) {}

        public RegistrationUpdate(string[] properties) : base(AthleteRaceStatus.Registered, properties)
        {
            if (properties.Length!=7 || properties[0] != AthleteRaceStatus.Registered.ToString())
                throw new ApplicationException("Invalid properties");

            FirstName = properties[3];
            LastName = properties[4];
            Gender = properties[5];
            Age = Convert.ToInt32(properties[6]);
        }


        public override string ToString()
        {
            return $"{base.ToString()},{FirstName},{LastName},{Gender},{Age}";
        }

    }
}

File: hw2/RaceMonitor_Java/src/Messages/RegistrationUpdate.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package Messages;

import Exceptions.ApplicationException;
import Racedata.AthleteRaceStatus;
/**
 *
 * @author Shimin
 * @author Stephen Clyde
 */
public class RegistrationUpdate extends AthleteUpdate {
    private String firstName;
    private String lastName;
    private String gender;
    private int age;
    
    public RegistrationUpdate() {
        super(AthleteRaceStatus.Registered);
    }
    
    public RegistrationUpdate(String[] properties) throws ApplicationException {
        super (AthleteRaceStatus.Registered, properties);
        
        if (properties.length != 7 || !properties[0].equals(AthleteRaceStatus.Registered.toString()))
            throw new ApplicationException("Invalid properties");

        firstName = properties[3];
        lastName = properties[4];
        gender = properties[5];
        age = Integer.parseInt(properties[6]);
    }
    
    @Override
    public String toString() {
        return super.toString() + "," + firstName + "," + lastName + "," + gender + "," + age;
    }
    
    public void setFirstName (String value) {
        firstName = value;
    }

    public String getFirstName () {
        return firstName;
    }
    
    public void setLastName (String value) {
        lastName = value;
    }

    public String getLastName () {
        return lastName;
    }
    
    public void setGender (String value) {
        gender = value;
    }

    public String getGender () {
        return gender;
    }

    public void setAge (int value) {
        age = value;
    }

    public int getAge () {
        return age;
    }
}
